## In Class Practice with Pandas

**Data Dictionary**

* Name - Name of the person
* Net worth in billions - Net worth of the person in billions
* Country - Country where the person is from
* Source - Source of the income
* Rank - Rank of the person in the Billionaires list
* Age - Age of the person
* Industry - Industry to which the person is related to

### Importing Pandas Library

In [1]:
import pandas as pd

### Loading the dataset in a pandas dataframe

In [46]:
df = pd.read_csv('Forbes-list-2020.csv')

### Print top 5 rows 

In [4]:
df.head()

Name  NetWorth in billions        Country  \
0                Jeff Bezos                 177.0  United States   
1                 Elon Musk                 151.0  United States   
2  Bernard Arnault & family                 150.0         France   
3                Bill Gates                 124.0  United States   
4           Mark Zuckerberg                  97.0  United States   

          Source  Rank  Age          Industry  
0         Amazon     1   57        Technology  
1  Tesla, SpaceX     2   49        Automotive  
2           LVMH     3   72  Fashion & Retail  
3      Microsoft     4   65        Technology  
4       Facebook     5   36        Technology

#### <span style="color:crimson">Q1. Whose name is assigned to the 0th index in the dataframe</span>
A/ Jeff Bezos

### Print the dimensions of data

In [5]:
df.shape

(2755, 7)

#### <span style="color:crimson">Q2. How many records do you have in this dataset?</span>
A/ 2755 records

### Datatypes of all the fields in the data

In [16]:
df.dtypes
#df.info()

Name                     object
NetWorth in billions    float64
Country                  object
Source                   object
Rank                      int64
Age                       int64
Industry                 object
dtype: object

#### <span style="color:crimson">Q3. What is the data type of the variable Source</span>
A/ object

### Checking for missing values

In [21]:
print(df.isnull().sum())
df.info()

Name                    0
NetWorth in billions    0
Country                 0
Source                  0
Rank                    0
Age                     0
Industry                0
dtype: int64
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2755 entries, 0 to 2754
Data columns (total 7 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Name                  2755 non-null   object 
 1   NetWorth in billions  2755 non-null   float64
 2   Country               2755 non-null   object 
 3   Source                2755 non-null   object 
 4   Rank                  2755 non-null   int64  
 5   Age                   2755 non-null   int64  
 6   Industry              2755 non-null   object 
dtypes: float64(1), int64(2), object(4)
memory usage: 150.8+ KB


#### <span style="color:crimson">Q4. How many missing values exists in the industry feature</span>
A/ there are not missing values

### Print the basic stats for this dataset

In [24]:
df.describe()

NetWorth in billions         Rank          Age
count           2755.000000  2755.000000  2755.000000
mean               4.749220  1345.663521    63.014519
std                9.615358   772.669811    13.259017
min                1.000000     1.000000    18.000000
25%                1.500000   680.000000    54.000000
50%                2.300000  1362.000000    62.000000
75%                4.200000  2035.000000    72.000000
max              177.000000  2674.000000    97.000000

#### <span style="color:crimson">Q5. Who is the yongest billionare in this list?</span>
A/ He is 18 years old, Kevin David Lehmann	

### Sort the dataset  by Age (ascending) and reset the index

In [47]:
df.sort_values("Age", ascending=True)
df.reset_index(drop=True)

Name  NetWorth in billions        Country  \
0                   Jeff Bezos                 177.0  United States   
1                    Elon Musk                 151.0  United States   
2     Bernard Arnault & family                 150.0         France   
3                   Bill Gates                 124.0  United States   
4              Mark Zuckerberg                  97.0  United States   
...                        ...                   ...            ...   
2750         Daniel Yong Zhang                   1.0          China   
2751             Zhang Yuqiang                   1.0          China   
2752             Zhao Meiguang                   1.0          China   
2753            Zhong Naixiong                   1.0          China   
2754           Zhou Wei family                   1.0          China   

             Source  Rank  Age          Industry  
0            Amazon     1   57        Technology  
1     Tesla, SpaceX     2   49        Automotive  
2              LVMH     3   72  Fashion & Retail  
3         Microsoft     4   65        Technology  
4          Facebook     5   36        Technology  
...             ...   ...  ...               ...  
2750     e-commerce  2674   49        Technology  
2751     Fiberglass  2674   65     Manufacturing  
2752    gold mining  2674   58   Metals & Mining  
2753   conglomerate  2674   58       Diversified  
2754       Software  2674   54        Technology  

[2755 rows x 7 columns]

In [48]:
#run
df.iloc[1200,3]

'real estate'

#### <span style="color:crimson">Q6. What is the output of `df.iloc[1200,3]`?</span>
A/ real state

### Create a new column `Year_of_birth`
Note: The dataset is from the year 2020. The age of those Billionaires is as of 2020.

In [31]:
# create a new column 'Year_of_birth'
df["Year_of_birth"] = 2020 - df["Age"]

In [32]:
#view the data
df.head()

Name  NetWorth in billions        Country       Source  \
940   Kevin David Lehmann                   3.3        Germany   drugstores   
2143   Alexandra Andresen                   1.4         Norway  investments   
2122          Wang Zelong                   1.5          China    chemicals   
2144   Katharina Andresen                   1.4         Norway  investments   
1338       Austin Russell                   2.4  United States      Unknown   

      Rank  Age          Industry  Year_of_birth  
940    925   18  Fashion & Retail           2002  
2143  2141   24       Diversified           1996  
2122  2035   24   Metals & Mining           1996  
2144  2141   25       Diversified           1995  
1338  1299   26        Automotive           1994

### Select Billionaires born in 1996

In [42]:
# Using where condition
df.where(df["Year_of_birth"] == 1996, "")

Name NetWorth in billions Country       Source  Rank Age  \
940                                                                            
2143  Alexandra Andresen                  1.4  Norway  investments  2141  24   
2122         Wang Zelong                  1.5   China    chemicals  2035  24   
2144                                                                           
1338                                                                           
...                  ...                  ...     ...          ...   ...  ..   
729                                                                            
1579                                                                           
1626                                                                           
1559                                                                           
170                                                                            

             Industry Year_of_birth  
940                                  
2143      Diversified          1996  
2122  Metals & Mining          1996  
2144                                 
1338                                 
...               ...           ...  
729                                  
1579                                 
1626                                 
1559                                 
170                                  

[2755 rows x 8 columns]

#### <span style="color:crimson">Q7. How many Billionaires were born in 1996?</span>
a/ 2 billionaires

In [45]:
df.where(df["Year_of_birth"] == 1996).count()

Name                    2
NetWorth in billions    2
Country                 2
Source                  2
Rank                    2
Age                     2
Industry                2
Year_of_birth           2
dtype: int64